### PR TITLE
feat(docs): generate api.json from .d.ts file

### PR DESCRIPTION
**Pull request by cmichaelgraham**

### CODE REVIEW

File: build/tasks/doc.js [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
+var typedocExtractor = require("gulp-typedoc-extractor");
+var runSequence = require('run-sequence');
 
 gulp.task('doc-generate', function(){
-  return gulp.src(paths.source)
-    .pipe(yuidoc.parser(null, 'api.json'))
-    .pipe(gulp.dest(paths.doc));
+  return gulp.src([paths.output + '*.d.ts', paths.doc + '/core-js.d.ts', './jspm_packages/github/aurelia/*/*.d.ts'])
+    .pipe(typedoc({
+            target: "es6",
```
Comment (bryanrsmith) [inline build/tasks/doc.js]: Indentation looks off. It'd be great to use single quotes consistently (double quotes also appear in the `require`s above). It also looks like some tabs snuck in further down.

File: package.json [hunk 2]
```diff
@@ -63,6 +65,7 @@
     "require-dir": "^0.1.0",
     "run-sequence": "^1.0.2",
     "through2": "^2.0.0",
+    "vinyl": "^0.5.1",
```
Comment (bryanrsmith) [inline package.json]: This doesn't look like it's used directly. Is it needed?

**Comment by EisenbergEffect** [comment]
@cmichaelgraham Can you address these issues please? Thanks!

**Comment by cmichaelgraham** [comment]
i get this error if i remove vinyl - not sure why?

```
$ gulp doc
module.js:338
    throw err;
          ^
Error: Cannot find module 'vinyl'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (c:\a\au\router\node_modules\gulp\node_modules\gulp-util\index.js:2:9)

    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
```

**Comment by cmichaelgraham** [comment]
fixed the single quotes issue :+1:

**Comment by bryanrsmith** [comment]
@cmichaelgraham I think gulp-typescript-extractor needs to list vinyl as a dependency instead of a dev dependency. I think that's why it's not being installed right now.

**Pull request merged**